x-pack/test_utils/README.md

```diff
@@ -0,0 +1,255 @@
+than when unit testing single components in isolation.
+
+## Motivation
+
+We dediced (the Elasticsearch UI team) to build this small abstraction to help with our "client integration" testing. When testing complete "pages" 
+we get to test our application in a way that is closer to how a user would interact with it in a browser. It also gives us more confidence in 
+our tests and avoids testing implementation details. 
+
+We test everything up to the HTTP Requests made from the client to the Node.js API server. Those requets need to be mocked. This means that with a good 
+**API integration test** coverage of those  endpoints, we can reduce the functional tests to a minium.
+
+With this in mind, we needed a way to easily mount a component on a React Router `<Route>` (this component could possibily have _child_ routes and 
+need access to the browser URL parameters and query params). In order to solve that, the Testbed wraps the component around a `MemoryRouter`.
+
+On the other side, the majority of our current applications use Redux as state management so we needed a simple way to wrap our component under test 
+inside a redux store provider.
+
+## How to use it
+
+At the top of your test file (you only need to declare it once), register a new Testbed by providing a React Component an an optional configuration object. 
+You receive in return a function that you need to call to mount the component in your test.
+
+**Example 1**
+
+```ts
+// remote_clusters_list.helpers.ts
+
+import { registerTestBed } from '../../../../test_utils';
+import { RemoteClusterList } from '../../app/sections/remote_cluster_list';
+import { remoteClustersStore } from '../../app/store';
+import routing from '../../app/services/routing';
+
+const config = {
+  memoryRouter: {
+    onRouter(router) {
+      routing.registerRouter(router); // send the router instance to the routing service
+    },
+    initialEntries: ['/some-resource-name'], // the Router initial URL
+    componentRoutePath: '/:name' // the Component <Route /> path
+  },
+  store: remoteClusterStore
+};
+
+export const setup = registerTestBed(RemoteClusterList, config);
+```
+
+Once you have registered a TestBed, you can call the `setup()` function to mount the component in your tests. You will get an object 
+back with a set of utility methods to test the component (refer to the documentation below for a complete list of the methods available).
+
+```ts
+// remote_cluster.test.ts
+
+import { setup } from './remote_clusters_list.helpers.ts';
+
+describe('<RemoteClusterList />', () => {
+  test('it should have a table with 3 rows', () => {
+    const { component, exists, find } = setup();
+
+    // component is an Enzyme reactWrapper
+    console.log(component.debug());
+  
+    expect(exists('remoteClusterTable')).toBe(true);
+    expect(find('remoteClusterTable.row').length).toBe(3);
+  });
+});
+```
+
+## Chrome extension
```

Comment:
I think these instructions should mention that output appears in the Console, or the extension UI itself should mention that.

Reply:
Good point, I added that information.